uk.co.compucorp.civicrm.hrcomments/tests/phpunit/CRM/HRComments/BAO/CommentTest.php

```diff
@@ -76,6 +77,28 @@ public function testValidateParamsIsCalledOnCreate() {
     ]);
   }
 
+  /**
+   * @expectedException CRM_HRComments_Exception_InvalidCommentException
+   * @expectedExceptionMessage Comment can not be soft deleted during an update use the delete method instead!
+   */
+  public function testTryingToSoftDeleteCommentOnUpdateThrowsException() {
+    $comment1 = Comment::create([
+      'entity_id' => 1,
+      'entity_name' => 'LeaveRequest',
+      'text' => 'This is a sample comment',
+      'contact_id' => 1,
+    ]);
+
+    Comment::create([
+      'id' => $comment1->id,
+      'entity_id' => 1,
+      'entity_name' => 'LeaveRequest',
+      'text' => 'This is a sample comment',
+      'contact_id' => 1,
```

Comment:
Since all the other fields are the same, on this call you only need to pass the `id` and `is_deleted`